### DISCUSSION TITLE
I'm confuse what should be the scope of Keycloak

### DISCUSSION THREAD

**Discussion by kamontat** [keycloak/keycloak#23935] [Q&A]
## Usecase

Think as we are an E-commerce website. We have login system (using Keycloak) to add additional discount to small group of user. Later, I plan to build the settings page so user can settings their account (e.g. language, currency, or subscription).

My question is this kind of information (that related to user but may not related to identity or access) should be as a attributes of the user or should we create a different external service for managing this kind of information?

**Comment by jonkoops** [comment]
You can store this kind of information as attributes on the user. We're working on stabilizing [declarative user profile](https://www.keycloak.org/server/features#_preview_features), which will give you even more power over how these attributes are formatted and presented to the user.

**Reply by kamontat** [reply]
If we have legacy storage collect this kind of informations (with restAPIs support).
How can I fetch legacy data once user register to new system.
